### PR TITLE
refactor and dedupe concurrency logic

### PR DESCRIPTION
**Pull request by BenTheElder**
we use these in a few places, this kind of code tends to be error prone, so let's dedupe it.

### CODE REVIEW

**Comment by BenTheElder** [comment]
cc @aojea

**Comment by BenTheElder** [comment]
naming is hard, I wanted to name this package `go` so we could do `go.UntilError` but of course that's reserved, so `concurrent.UntilError` it is. perhaps we can come up with more precise names in a follow up 🙃

File: pkg/concurrent/concurrent.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// first non-nil error returned from funcs, or nil if all funcs return nil
+func UntilError(funcs []func() error) error {
+	errCh := make(chan error, len(funcs))
+	for _, f := range funcs {
+		f := f // capture f
+		go func() {
+			errCh <- f()
+		}()
+	}
+	for i := 0; i < len(funcs); i++ {
```
Comment (aojea) [inline pkg/concurrent/concurrent.go]: just curious, this `for` loop is the one that waits for all goroutines to finish, isn´t it?

Comment (BenTheElder) [reply pkg/concurrent/concurrent.go]: UntilError returns the first error immediately without waiting.

Comment (aojea) [reply pkg/concurrent/concurrent.go]: ic, and if no errors it waits for all functions to send the err results. Thanks

Comment (BenTheElder) [reply pkg/concurrent/concurrent.go]: right, we can't return `nil` correctly until every func has returned, but we can return an error immediately, this one is meant for quick bail-out (EG for cluster creation code), while we have `Coalesce` for when we want to continue and collect the errors (as in collecting logs).

**Pull request merged**